src/components/Facility/BedCapacity.tsx

```diff
@@ -201,7 +175,11 @@ export const BedCapacity = (props: BedCapacityProps) => {
         }
         handleUpdate();
       }
-      if (btnType == "Save and Exit") handleClose();
+      if (
+        btnType !== "save-and-add-more" ||
+        bedTypes.length === BED_TYPES.length
```

Comment:
The latter comparison wouldn't work as `bedTypes` length will always be the same.

Here, you want to bring `updatedBedTypes` out of scope earlier*, calculate the number of beds that are disabled and then do a check (if it exists + if it equals to `BED_TYPES` length).

Something like:

`const disabledBedTypesLength = updatedBedTypes?.filter((item) => item.disabled).length;`

*In this case, we can't use `bedTypes` right away since we are updating it (state may or may not be updated in time).

Reply:
Got it.